tests/core/data_model/streams/test_data_stream.py

```diff
@@ -17,16 +17,31 @@
 import os
 import pickle
 
+# Third Party
+import pytest
+
 # Local
-from caikit.core import data_model as core_dm
 from caikit.core.augmentors import AugmentorBase
+from caikit.core.data_model import DataStream
 from sample_lib.data_model.sample import SampleInputType, SampleOutputType
-from sample_lib.modules.sample_task.sample_implementation import SampleModule
-
-# Unit Test Infrastructure
-from tests.base import TestCaseBase
 import caikit.core
 
+###############################
+# Helper functions and fixtures
+###############################
+
+
+def validate_data_stream(data_stream, length, data_item_type, data_item_length=None):
+    # Verify lengths and types
+    assert isinstance(data_stream, DataStream)
+    assert len(data_stream) == length
+    assert sum(1 for _ in data_stream) == length
```

Comment:
I've never understood why we did this instead of `len(list(data_stream)) == length`

Reply:
We found out the hard way- it's to test re-entrance. The initial `len` call has to iterate over the entire stream, then the `sum` will fail if the stream isn't re-entrant

Reply:
Ah- but yeah we probably could do that with a `len(list(stream))` ? But that might hold the whole stream in memory and some nefarious person could put a 50GB stream in the unit tests 😉

Reply:
Fair enough!